### PR TITLE
Handle missing effort limit in URDF

### LINKED ISSUE #1083: Effort display seg fault when no effort limit is specified in URDF
https://github.com/ros2/rviz/pull/990 by @ahcorde introduced Effort display.
However in these lines:
https://github.com/ros2/rviz/blob/01df29780a7d90917b9d927bf09a931d1e57f223/rviz_default_plugins/src/rviz_default_plugins/displays/effort/effort_display.cpp#L197-L200
`limit` is not checked and might be `nullptr`.
This IMHO shall be handled: the user shall get a warning that the joint has no associated effort limit in URDF but this is required for visualization. Maybe the joint can be skipped (i.e. not inserted in the tree in the plugin sidebar).

### PR DESCRIPTION
**Pull request by roncapat**
Fixes #1083

### CODE REVIEW

File: rviz_default_plugins/src/rviz_default_plugins/displays/effort/effort_display.cpp [hunk 1]
```diff
@@ -196,8 +196,16 @@ void EffortDisplay::topic_callback(const std_msgs::msg::String & msg)
...
       urdf::JointLimitsSharedPtr limit = joint->limits;
-      joints_[joint_name] = std::make_shared<JointInfo>(joint_name, joints_category_);
-      joints_[joint_name]->setMaxEffort(limit->effort);
+      if (limit) {
+        joints_[joint_name] = std::make_shared<JointInfo>(joint_name, joints_category_);
+        joints_[joint_name]->setMaxEffort(limit->effort);
+      } else {
+        RCLCPP_WARN(
+          context_->getRosNodeAbstraction().lock()->get_raw_node()->get_logger(),
+          "Joint `%s` has no <limit> tag in URDF. Effort plugin needs to know the effort "
```
Comment (clalancette) [inline rviz_default_plugins/src/rviz_default_plugins/displays/effort/effort_display.cpp]:
```suggestion
          "Joint '%s' has no <limit> tag in URDF. Effort plugin needs to know the effort "
```

(it is just more similar to what we do elsewhere)

File: rviz_default_plugins/src/rviz_default_plugins/displays/effort/effort_display.cpp [hunk 2]
```diff
@@ -196,8 +196,16 @@ void EffortDisplay::topic_callback(const std_msgs::msg::String & msg)
     if (joint->type == urdf::Joint::REVOLUTE || joint->type == 2) {
       std::string joint_name = it->first;
       urdf::JointLimitsSharedPtr limit = joint->limits;
-      joints_[joint_name] = std::make_shared<JointInfo>(joint_name, joints_category_);
-      joints_[joint_name]->setMaxEffort(limit->effort);
+      if (limit) {
+        joints_[joint_name] = std::make_shared<JointInfo>(joint_name, joints_category_);
+        joints_[joint_name]->setMaxEffort(limit->effort);
+      } else {
+        RCLCPP_WARN(
```
Comment (clalancette) [inline rviz_default_plugins/src/rviz_default_plugins/displays/effort/effort_display.cpp:204]: I'll suggest you add `#include <rclcpp/logging.hpp>` to the top of this file.

**Review by clalancette — CHANGES_REQUESTED** [review]
A couple of small things to fix, then I'm happy with this.

**Review by clalancette — APPROVED** [review]
Looks good to me!  I'll run CI on it next.

**Comment by clalancette** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/19746/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/14246/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/20468/)

**Comment by ahcorde** [comment]
https://github.com/Mergifyio backport humble iron

**Pull request merged**